aspnet-core/src/Timesheet.Application/APIs/ReviewDetails/ReviewDetailAppService.cs

```diff
@@ -1446,5 +1477,55 @@ public async Task<List<InternshipMaxLevelMonthsDto>> CountMonthLevelMax()
                         .ToListAsync();
             return listInternshipMaxLevelMonths;
         }
+
+        public bool CheckRecordRemainingWithStatusAndId(long reviewId, ReviewInternStatus status, List<long> listId)
+        {
+            int count = WorkScope.GetAll<ReviewDetail>()
+                .Where(x => x.Status == status && x.ReviewId == reviewId && !listId.Contains(x.Id)).Count();
+            return count == 0;
+        }
+
+        public bool CheckRecordRemainingWithStatus(long reviewId, ReviewInternStatus status)
+        {
+            return WorkScope.GetAll<ReviewDetail>()
+                .Where(x => x.ReviewId == reviewId)
+                .All(x => x.Status == status);
+        }
+
+        public async void SendMailToNotifyTransition(string email, string username, ReviewInternStatus status, long reviewId)
+        {
+            ReviewIntern reviewIntern = await WorkScope.GetAsync<ReviewIntern>(reviewId);
+            StringBuilder content = new StringBuilder("");
+            try
+            {
+                if (status == ReviewInternStatus.Reviewed)
+                {
+                    content.Append($"Thân gửi anh <span style='font-weight: 600'>{username} </span>, <br> Head PM đã đánh giá xong.");
+                } 
+                else
+                {
+                    content.Append($"Thân gửi anh <span style='font-weight: 600'>{username} </span>, <br> Các PM đã đánh giá xong thực tập sinh");
+                }
+                content.Append("<br>");
+                content.Append("Mời anh vào đánh giá ạ.");
+                string link = "https://timesheet.nccsoft.vn/app/main/review-detail?id=" + reviewIntern.Id + "&year=" + reviewIntern.Year + "&month=" + reviewIntern.Month;
```

Comment:
should get domain from config cause we have multi env
this one timesheet.nccsoft.vn for prod only